### PR TITLE
KAFKA-13983: Fail the creation with "/" in resource name in zk ACL

### PR DESCRIPTION
**Pull request by singhnama**
[Problem](https://issues.apache.org/jira/browse/KAFKA-13983)
Sanitizing the resource name have a compatibility issue with already existing ACLs, and also there are very few special characters which are not allowed by the zookeeper which can be found [here](https://zookeeper.apache.org/doc/r3.1.2/zookeeperProgrammers.html) so we decided to fail the creation with `/` in the resource name since this is allowed by zookeeper but it's creating the problem in acl operation.

Not allowed characters by zookeeper:

```
The null character (\u0000) cannot be part of a path name. (This causes problems with the C binding.)

The following characters can't be used because they don't display well, or render in confusing ways: \u0001 - \u0019 and \u007F - \u009F.

The following characters are not allowed: \ud800 -uF8FFF, \uFFF0-uFFFF, \uXFFFE - \uXFFFF (where X is a digit 1 - E), \uF0000 - \uFFFFF.

The "." character can be used as part of another name, but "." and ".." cannot alone be used to indicate a node along a path, because ZooKeeper doesn't use relative paths. The following would be invalid: "/a/b/./c" or "/a/b/../c".

The token "zookeeper" is reserved.

```

### CODE REVIEW

**Review by hachikuji — COMMENTED** [review]
Thanks for the fix. Are there any test cases we can add?

**Comment by singhnama** [comment]
Sanitizing the resource name have a compatibility issue with already existing ACLs,  and also there are very few special characters which are not allowed by the zookeeper which can be found here [https://zookeeper.apache.org/doc/r3.1.2/zookeeperProgrammers.html](https://zookeeper.apache.org/doc/r3.1.2/zookeeperProgrammers.html) so we decided to fail the creation with `/` in the resource name since this is allowed by zookeeper but it's creating the problem.

Not allowed characters:
````
The null character (\u0000) cannot be part of a path name. (This causes problems with the C binding.)

The following characters can't be used because they don't display well, or render in confusing ways: \u0001 - \u0019 and \u007F - \u009F.

The following characters are not allowed: \ud800 -uF8FFF, \uFFF0-uFFFF, \uXFFFE - \uXFFFF (where X is a digit 1 - E), \uF0000 - \uFFFFF.

The "." character can be used as part of another name, but "." and ".." cannot alone be used to indicate a node along a path, because ZooKeeper doesn't use relative paths. The following would be invalid: "/a/b/./c" or "/a/b/../c".

The token "zookeeper" is reserved.
````

File: core/src/main/scala/kafka/security/authorizer/AclAuthorizer.scala [hunk 1]
```diff
@@ -209,6 +213,9 @@ class AclAuthorizer extends Authorizer with Logging {
           throw new UnsupportedVersionException(s"Adding ACLs on prefixed resource patterns requires " +
             s"${KafkaConfig.InterBrokerProtocolVersionProp} of $IBP_2_0_IV1 or greater")
         }
+        if (inValidAclBindingResourceName(aclBinding.pattern().name())) {
```
Comment (omkreddy) [inline core/src/main/scala/kafka/security/authorizer/AclAuthorizer.scala]: Can we this to `validateAclBinding` method?

Comment (singhnama) [reply core/src/main/scala/kafka/security/authorizer/AclAuthorizer.scala]: updated

**Review by omkreddy — APPROVED** [review]
@singhnama Thanks for the PR. LGTM.

**Pull request merged**